signing.go

```diff
@@ -0,0 +1,105 @@
+package matrixfederation
+
+import (
+	"encoding/json"
+	"fmt"
+	"golang.org/x/crypto/ed25519"
+)
+
+// SignJSON signs a JSON object returning a copy signed with the given key.
+// https://matrix.org/docs/spec/server_server/unstable.html#signing-json
+func SignJSON(entityName, keyID string, privateKey ed25519.PrivateKey, message []byte) ([]byte, error) {
```

Comment:
Please stick to the same terminology for variable names as is used in the spec. That is:
 - `s/keyID/signingKeyID/`
 - `s/entityName/signingName/`